system_tests/logging_.py

```diff
@@ -50,8 +50,18 @@ def setUp(self):
         self.to_delete = []
 
     def tearDown(self):
+        from gcloud.exceptions import NotFound
         for doomed in self.to_delete:
-            doomed.delete()
+            backoff_intervals = [1, 2, 4, 8]
+            while True:
+                try:
+                    doomed.delete()
+                    break
+                except NotFound:
```

Comment:
Have you seen this occur in the wild? Why would a `NotFound` occur on something that needs to be deleted? Also, wouldn't it occur once the delete succeeds?